### ISSUE TITLE
Trainer.fit reads len(data) as the dataset size, so minibatch likelihoods get the wrong scale

### ISSUE THREAD

**Issue by YichengYang-Ethan** [pymc-devs/pymc-extras#758]
`Trainer.fit` takes `len(data)` as N, but for an iterable of batches `len` is the number of batches, not rows. So the likelihood gets scaled by batches/batch_rows instead of N/batch_rows, and nothing warns.

No DataLoader needed to hit it:

```python
import numpy as np, pymc as pm
from pymc_extras.inference.advi.training import Trainer

y = np.random.default_rng(0).normal(1.5, 1.0, 3000)
batches = [{"y": y[i:i+100]} for i in range(0, 3000, 100)]   # 30 batches, 3000 rows

with pm.Model() as m:
    mu = pm.Normal("mu", 0, 5)
    d = pm.Data("y", batches[0]["y"])
    pm.Normal("y_obs", mu, 1.0, observed=d)

t = Trainer()
t.fit(30, data=batches, model=m, random_seed=1)
print(t._logp_scalings)      # {'y_obs': 0.3}, expected 30.0
```

`len` being the batch count is the convention from e386f592, which also says to use `total_size` for the PyMC parameter. `test_fit_rescales_likelihood_when_stream_has_len` uses a loader whose `__len__` returns N, so this passes CI.

Same code path, second thing: the scale is computed once from the first batch in `_prepare_data_stream` and compiled in as a constant. With a ragged first batch (37 rows, then 100s, N=3000) every later batch keeps running at 3000/37, so effective N is 8108. Also silent.

pymc-extras d2591e1, pymc 6.2.0, clean venv.

Rough idea of a fix, but I'd rather hear how you want it:

- leave RVs that already declare `total_size` alone, since that's per-RV and can be multi-axis
- otherwise take N from an explicit `fit(total_size=...)`, then `data.total_size`; raise if both are there and disagree, warn if neither
- for the ragged half, compile the scale as `N / value.shape[0]` the way `MinibatchRandomVariable` does

Two things I'm not sure about. I tried a `len(data) * batch_rows ≈ N` cross-check and dropped it, since the test above breaks it (replacement sampling) and so would ragged batches or a distributed loader. And `DataLoader` yields arrays while `fit` wants dicts, so I tested through an adapter — there's no real end-to-end test yet.
